include/deal.II/lac/trilinos_precondition.h

```diff
@@ -1382,6 +1385,101 @@ namespace TrilinosWrappers
                      const char *       smoother_type    = "Chebyshev",
                      const char *       coarse_type      = "Amesos-KLU");
 
+      /**
+       * Constructor. By default, we pretend to work on elliptic problems with
+       * linear finite elements.
+       *
+       * This constructor initializes the @p constant_modes for the @p dof_handler
+       * and the components chosen by @p component_mask . See the documentation
+       * of the @p constant_modes parameter and
+       * DoFTools::extract_constant_modes() function for more information on
+       * this point.
+       */
+      template <typename DoFHandlerType>
+      AdditionalData(const DoFHandlerType &dof_handler,
+                     const ComponentMask & component_mask,
```

Comment:
I am not sure how we would like to handle this case, but it feels that this change goes somewhat against our directory philosophy of not including any FE-specific stuff in a linear algebra class. We probably won't ever do that, but if we wanted to break out the `base/` and `lac/` directories we could still manage that (we only have a handful of dependencies in `base`), but every change like this one would make it more difficult. How are the opinions on this?

Reply:
I wasn't conscious of the fact that we tried to keep these separate.  I can just drop this commit to keep things moving...